### ISSUE TITLE
terragrunt-strict-validate option is not working as expected

### ISSUE THREAD

**Issue by tstraley** [gruntwork-io/terragrunt#1793]
terragrunt version v0.31.6

When running [validate-inputs](https://terragrunt.gruntwork.io/docs/reference/cli-options/#validate-inputs) with the strict validate flag, it does not error on unused inputs like it claims it should. Only warnings. Running in debug mode shows that it is not enabling strict checking.

```shell
$ terragrunt validate-inputs --terragrunt-strict-validate
WARN[0000] The following inputs passed in by terragrunt are unused:  prefix=[/host_home/Projects/infra] 
WARN[0000] 	- foo                                        prefix=[/host_home/Projects/infra] 
WARN[0000]                                               prefix=[/host_home/Projects/infra] 
INFO[0000] All required inputs are passed in by terragrunt  prefix=[/host_home/Projects/infra] 
WARN[0000] Terragrunt configuration has misaligned inputs, but running in relaxed mode so ignoring.  prefix=[/host_home/Projects/infra] 


$ terragrunt validate-inputs --terragrunt-strict-validate --terragrunt-log-level debug
DEBU[0000] Found locals block: evaluating the expressions. 
DEBU[0000] Evaluated 5 locals (remaining 0): aws_region, aws_profile, aws_acct_name, aws_acct_id, env_name 
DEBU[0000] Running command: terraform --version          prefix=[/host_home/Projects/infra] 
DEBU[0000] Terraform version: 1.0.5                     
DEBU[0000] Reading Terragrunt config file at /host_home/Projects/infra/terragrunt.hcl 
DEBU[0000] Found locals block: evaluating the expressions. 
DEBU[0000] Evaluated 5 locals (remaining 0): aws_acct_id, env_name, aws_region, aws_profile, aws_acct_name 
DEBU[0000] Downloading Terraform configurations from file:///host_home/Projects/infra/terraform into /host_home/Projects/infra/.terragrunt-cache/7U3u4b5Qruzi8IXcDWJKVhY1rpU/L76ZjGDuhKIWoP-Ukd4lmxYLxz8 
DEBU[0000] Copying files from /host_home/Projects/infra into /host_home/Projects/infra/.terragrunt-cache/7U3u4b5Qruzi8IXcDWJKVhY1rpU/L76ZjGDuhKIWoP-Ukd4lmxYLxz8 
DEBU[0000] Setting working directory to /host_home/infra/.terragrunt-cache/7U3u4b5Qruzi8IXcDWJKVhY1rpU/L76ZjGDuhKIWoP-Ukd4lmxYLxz8 
WARN[0000] The following inputs passed in by terragrunt are unused:  prefix=[/host_home/Projects/infra] 
WARN[0000] 	- foo                                        prefix=[/host_home/Projects/infra] 
WARN[0000]                                               prefix=[/host_home/Projects/infra] 
INFO[0000] All required inputs are passed in by terragrunt  prefix=[/host_home/Projects/infra] 
DEBU[0000] Strict mode enabled: false                    prefix=[/host_home/Projects/infra] 
WARN[0000] Terragrunt configuration has misaligned inputs, but running in relaxed mode so ignoring.  prefix=[/host_home/Projects/infra] 

$ echo $?
0
```

**Comment by brikis98** [comment]
That does look like a bug. Do you have any easy test case we can use to repro?

**Comment by davidorlea** [comment]
Any updates here? Since missing variables will already be detected by failing plans and applies, `validate-inputs` without `terragrunt-strict-validate` is almost of no use.

**Comment by denis256** [comment]
Fixed in release v0.36.10

https://github.com/gruntwork-io/terragrunt/releases/tag/v0.36.10

**Comment by oatmealb** [comment]
Apologies if this should've been a new issue, but at least the existing title fits perfectly to my issue: a `` call passes, yet `terragrunt apply` fails. Here's the shortened output:

```sh
$ terragrunt validate-inputs --terragrunt-strict-validate && terragrunt apply 
INFO[0009] All variables passed in by terragrunt are in use.  prefix=[/full/path/] 
INFO[0009] All required inputs are passed in by terragrunt  prefix=[/full/path] 
Acquiring state lock. This may take a few moments...

│ Error: Missing required argument
│ ...
│ The argument "foo" is required, but no definition was found.
```
I'll try to come up with a repro. But firstly, should the above even be possible? Or am i misunderstanding the docs:
>    terragrunt-strict-validate                   Sets strict mode for the validate-inputs command. By default, strict mode is off. When this flag is passed, strict mode is turned on. When strict mode is turned off, the validate-inputs command will only return an error if required inputs are missing from all input sources (env vars, var files, etc). When strict mode is turned on, an error will be returned if required inputs are missing OR if unused variables are passed to Terragrunt.
